src/settings.rs

```diff
@@ -0,0 +1,67 @@
+use clap::{crate_authors, crate_description, crate_version, load_yaml, App};
+use config::{Config, ConfigError, File};
+use serde::Deserialize;
+
+const DEFAULT_LOTUSD_URL: &str = "http://127.0.0.1:7632";
+const DEFAULT_LOTUSD_USER: &str = "lotus";
+const DEFAULT_LOTUSD_PASSWORD: &str = "lotus";
+const FOLDER_DIR: &str = ".lotus-gpu-miner";
+
+#[derive(Debug, Deserialize)]
+pub struct Settings {
+    pub lotusd_url: String,
+    pub lotusd_user: String,
+    pub lotusd_password: String,
+    pub output_address: String,
+}
+
+impl Settings {
+    pub fn new() -> Result<Self, ConfigError> {
+        let mut s = Config::new();
+
+        // Set defaults
+        let yaml = load_yaml!("cli.yaml");
+        #[allow(deprecated)]
```

Comment:
huh? is it hard to use non-deprecated code?

You can also use the newer `struct` way of defining the options. You define a `struct` and annotate the fields with the required clap metadata for the options.